### PR TITLE
networkd: VXLAN rename Id to VNI

### PR DESCRIPTION
**Pull request by ssahani**
It makes more sense to call VXLAN ID as the VXLAN VNI Network Identifier (or VXLAN Segment ID)

### CODE REVIEW

**Comment by keszybz** [comment]
Is VNI short for virtual network identifier?

**Comment by ssahani** [comment]
> Is VNI short for virtual network identifier?

It's VXLAN Network Identifier or VNI

File: src/network/netdev/vxlan.c [hunk 1]
```diff
@@ -273,8 +273,8 @@ static int netdev_vxlan_verify(NetDev *netdev, const char *filename) {
         assert(v);
         assert(filename);
 
-        if (v->id > VXLAN_VID_MAX) {
-                log_warning("VXLAN without valid Id configured in %s. Ignoring", filename);
+        if (v->vni > VXLAN_VID_MAX) {
+                log_warning("VXLAN without valid VNI Network Identifier (or VXLAN Segment ID) configured in %s. Ignoring", filename);
```
Comment (keszybz) [inline src/network/netdev/vxlan.c]: "VXLAN without valid VNI (or VXLAN Segment ID) configured ..."

File: man/systemd.netdev.xml [hunk 2]
```diff
@@ -527,9 +527,9 @@
 
     <variablelist class='network-directives'>
       <varlistentry>
-        <term><varname>Id=</varname></term>
+        <term><varname>VNI=</varname></term>
         <listitem>
-          <para>The VXLAN ID to use.</para>
+          <para>The VXLAN VNI Network Identifier (or VXLAN Segment ID). Ranges [0-16777215].</para>
```
Comment (keszybz) [inline man/systemd.netdev.xml]: ... so "VXLAN VNI Network Identifier" expands to "VXLAN VXLAN Network Identifier Network Identifier". Please just shorten this to "VXLAN Network Identifier" here.

Comment (ssahani) [reply man/systemd.netdev.xml]: Yes updated . thanks.

File: man/systemd.netdev.xml [hunk 3]
```diff
@@ -527,9 +527,9 @@
 
     <variablelist class='network-directives'>
       <varlistentry>
-        <term><varname>Id=</varname></term>
+        <term><varname>VNI=</varname></term>
         <listitem>
-          <para>The VXLAN ID to use.</para>
+          <para>The VXLAN Network Identifier (or VXLAN Segment ID). Ranges [0-16777215].</para>
```
Comment (keszybz) [inline man/systemd.netdev.xml]: A number in the range 0..16777215.

Comment (ssahani) [reply man/systemd.netdev.xml]: How about "  A number ranges [1-16777215]."
BTW we have similar directives will update them as well later.

Comment (yuwata) [reply man/systemd.netdev.xml]: I prefer `Takes a number in the range [1-16777215].` or `Takes a number between 1 and 16777215.`

File: src/network/netdev/vxlan.c [hunk 4]
```diff
@@ -176,11 +182,11 @@ int config_parse_vxlan_address(const char *unit,
 
         if (streq(lvalue, "Group")) {
                 if (r <= 0) {
-                        log_syntax(unit, LOG_ERR, filename, line, 0, "vxlan %s invalid multicast address, ignoring assignment: %s", lvalue, rvalue);
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "vxlan Group configured with invalid multicast address, ignoring assignment: %s", rvalue);
```
Comment (yuwata) [inline src/network/netdev/vxlan.c]: Maybe, `VXLAN Group`? `vxlan Group` looks strange for me...

File: src/network/netdev/vxlan.c [hunk 5]
```diff
@@ -273,11 +279,24 @@ static int netdev_vxlan_verify(NetDev *netdev, const char *filename) {
...
+                return -EINVAL;
+        }
+
+        if (!in_addr_is_null(v->group_family, &v->group) && !in_addr_is_null(v->remote_family, &v->remote)) {
+                log_warning("VXLAN configured with both Group and Remote in %s. Ignoring", filename);
                 return -EINVAL;
         }
 
+        if (v->ttl > 255) {
+                log_warning("VXLAN TTL must be <= 255 in %s. Ignoring", filename);
```
Comment (yuwata) [inline src/network/netdev/vxlan.c]: `VXLAN TTL must be <= 255. Ignoring %s.` or `Ignoring: %s`.

File: src/network/netdev/vxlan.c [hunk 6]
```diff
@@ -27,17 +27,17 @@ static int netdev_vxlan_fill_message_create(NetDev *netdev, Link *link, sd_netli
...
-                r = sd_netlink_message_append_u32(m, IFLA_VXLAN_ID, v->id);
+        if (v->vni <= VXLAN_VID_MAX) {
+                r = sd_netlink_message_append_u32(m, IFLA_VXLAN_ID, v->vni);
                 if (r < 0)
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_VXLAN_ID attribute: %m");
         }
 
-        if (in_addr_is_null(v->remote_family, &v->remote) == 0) {
+        if (in_addr_is_null(v->group_family, &v->group) == 0) {
                 if (v->remote_family == AF_INET)
```
Comment (yuwata) [inline src/network/netdev/vxlan.c]: Not sure, but maybe `v->group_family`?

**Comment by ssahani** [comment]
Updated thanks for the review @keszybz and @yuwata .

File: man/systemd.netdev.xml [hunk 7]
```diff
@@ -529,7 +529,7 @@
       <varlistentry>
         <term><varname>Id=</varname></term>
         <listitem>
-          <para>The VXLAN ID to use.</para>
+          <para>The VXLAN ID to use. Takes a number in the range [1-16777215].</para>
```
Comment (keszybz) [inline man/systemd.netdev.xml]: You really don't need braces here. Also use either an n-dash or dots.

File: man/systemd.netdev.xml [hunk 8]
```diff
@@ -688,7 +688,7 @@
       <varlistentry>
         <term><varname>Id=</varname></term>
         <listitem>
-          <para>Specifies the Virtual Network Identifier (VNI) to use. Ranges [0-16777215].</para>
+          <para>Specifies the Virtual Network Identifier (VNI) to use. Takes a number in the range [0-16777215].</para>
```
Comment (keszybz) [inline man/systemd.netdev.xml]: The same ehre.

File: src/network/netdev/vxlan.c [hunk 9]
```diff
@@ -284,6 +284,11 @@ static int netdev_vxlan_verify(NetDev *netdev, const char *filename) {
                 return -EINVAL;
         }
 
+        if (v->ttl > 255) {
+                log_warning("VXLAN TTL must be <= 255. Ignoring %s.", filename);
+                return -EINVAL;
+        }
```
Comment (keszybz) [inline src/network/netdev/vxlan.c]:
```C
return log_warning_errno(SYNTHETIC_ERRNO(EINVAL), ...);
```

File: man/systemd.netdev.xml (hunk 7 above)
Comment (yuwata) [reply man/systemd.netdev.xml]: I like the `[1-nnn]` format. It is mathematically clear.

I cannot remember but have we discussed that in somewhere?

Comment (ssahani) [reply man/systemd.netdev.xml]: Yes much better but updated now according to @keszybz .

**Comment by ssahani** [comment]
updated thanks for review @yuwata and @keszybz .

File: man/systemd.netdev.xml (hunk 7 above)
Comment (yuwata) [reply man/systemd.netdev.xml]: OK. We have used all `[]`, `..`, `-` in many places. Let's cleanup in another PR.

**Comment by yuwata** [comment]
I've updated one more `log_warning` -> `log_warning_errno`.

File: man/systemd.netdev.xml (hunk 7 above)
Comment (keszybz) [reply man/systemd.netdev.xml]: I know of the following syntaxes:
- [a, b] or [a, b), or (a, b] — standard math notation for left-right-inclusive, left-inclusive, right-inclusive ranges
- a – b (note that it's an n-dash in there)
- a .. b (used e.g. by bash)

In standard math I'd interpret `[a-b]` as the floor of the difference of a and b.
But I agree that we can have the discussion in a separate PR.

**Comment by keszybz** [comment]
Thanks for the updates. Looks nice now.

**Comment by keszybz** [comment]
travis-ci fails with "Looks like test/fuzz/fuzz-netdev-parser/directives.netdev hasn't been updated".

**Comment by keszybz** [comment]
I turns out Group= was already in the corpus, which was causing the check to fail. I pushed two more commits on top.

**Comment by keszybz** [comment]
`i386` is failing in 24-UNIT-TESTS with a timeout:
Assertion 'sd_bus_start(bus) >= 0' failed at ../src/libsystemd/sd-bus/test-bus-watch-bind.c:140, function thread_client1(). Aborting.

I'll restart the test.

**Comment by ssahani** [comment]
Thanks @keszybz .
